docs/yaml/functions/summary.yaml

```diff
@@ -62,10 +62,30 @@ example: |
   ```
 
 posargs:
-  key:
-    type: str
-    description: The name of the new entry
+  key_or_dict:
+    type: str | dict
+    description: |
+      The name of the new entry, or a dictionary with values of the same type
+      as the `value` optarg.
```

Comment:
This is confusing and makes it sound like the value optarg is actually passed in this case. Also it's a bit indirect.

All dicts need a string key and the reference manual usually specifies the allowed values alone. So this could be `str | dict[str | bool | int | dep | external_program | list[str | bool | int | dep | external_program]]`